### PR TITLE
Make the "head only" check a separate formula audit

### PR DESCRIPTION
**Pull request by rickmark**
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

Not all taps have hard and fast rules on no head-only formulas.  By separating this audit type, it can be skipped explicitly without loosing the remainder of the `audit_spec` auditor.

### CODE REVIEW

**Comment by rickmark** [comment]
The intent here is to allow for `brew audit --skip=head_only` for our taps

File: Library/Homebrew/formula_auditor.rb [hunk 1]
```diff
@@ -560,9 +560,11 @@ def get_repo_data(regex)
       [user, repo]
     end
 
-    def audit_specs
+    def audit_head_only
       problem "Head-only (no stable download)" if head_only?(formula)
```
Comment (MikeMcQuaid) [inline Library/Homebrew/formula_auditor.rb:564]: I'd actually be fine with moving this below a `next unless @core_tap` but before a `next unless @new_formula` so it only ever applies for homebrew/core. Thoughts?

Comment (rickmark) [reply Library/Homebrew/formula_auditor.rb:564]: even better - sure its a good convention for core - but most non core stuff is .... lets say more bleeding edge.

I think we should be careful to allow one to re-enable the head only checkin they want to follow the convention?

Comment (MikeMcQuaid) [reply Library/Homebrew/formula_auditor.rb:564]: @rickmark Nah, I think it's fine to just make it core-only, thanks!

**Pull request closed without merging**